cadc-pkg-server/src/main/java/org/opencadc/pkg/server/PackageRunner.java

```diff
@@ -241,31 +233,49 @@ private void doIt() {
     }
 
     /**
-     * Initialize syncOutput output stream with the correct content type and disposition,
-     * as provided by the writer class
-     * @param packageName - package name to use
+     * Set up PackageWriter and syncoutput for the requested RESPONSEFORMAT value.
+     * Initialize syncOutput output stream with the correct content type and disposition
+     * as provided by the writer class. Call writer ctor.
+     * @param packageName
+     * @return
      * @throws IOException
      */
-    private void initOutputStream(String packageName) throws IOException {
+    private PackageWriter initWriter(String packageName) throws IllegalArgumentException, IOException {
 
         if (!StringUtil.hasText(packageName)) {
-            throw new RuntimeException("BUG: packageName can't be null");
+            throw new IllegalArgumentException("packageName not provided");
         }
 
-        syncOutput.setResponseCode(200);
+        // Package type is in RESPONSEFORMAT Job parameter
+        String responseFormat = ParameterUtil.findParameterValue("RESPONSEFORMAT", job.getParameterList());
+
+        if (!StringUtil.hasLength(responseFormat)) {
+            throw new IllegalArgumentException("RESPONSEFORMAT not found in job");
+        } else {
```

Comment:
Maybe "Missing required parameter: RESPONSEFORMAT", as this error will be seen by clients.

Reply:
done.